storagedriver/azure/blockid.go

```diff
@@ -0,0 +1,56 @@
+package azure
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"sync"
+
+	azure "github.com/MSOpenTech/azure-sdk-for-go/clients/storage"
+)
+
+type blockIDGenerator struct {
+	pool map[string]bool
+	m    sync.Mutex
+}
+
+// Generate returns an unused random block id and adds the generated ID
```

Comment:
We do need to keep track because if a blob consists of blocks `A`, `B`, `C`, `D` and if we're gonna append a new block, we shouldn't call if for example `B`, because if we do, blob will look like `A-B'-C-D-B'`.

And block IDs do last forever in the scope of that blob. Therefore we should name new blocks with strings that are not already existing or being used in this session.